src/nxdoc/nuxeo-add-ons/nuxeo-management-rest-api.md

```diff
@@ -0,0 +1,762 @@
+---
+title: 'Nuxeo Management REST API'
+review:
+    comment: ''
+    date: '2019-07-26 16:26'
+    status: ok
+labels:
+    - lts2019-ok    
+toc: true
+tree_item_index: 300
+history:
+---
+
+The Nuxeo Management REST API is a set of endpoints allowing the management of the Nuxeo Platform.
+
+This page explains these endpoints and their purpose.
+
+## Installation
+
+{{{multiexcerpt 'MP-installation-easy' page='Generic Multi-Excerpts'}}}
+
+## Configuration
+### Authentication
+
+The Management REST API is accessible for any administrator user.
+
+Furthermore, a "technical" user can be configured to access the Management REST API in nuxeo.conf:
+
+```
+org.nuxeo.rest.management.user=transient/technical_user
+```
+
+The user does not need to exist in Nuxeo, and **must** start with `transient/` as we are relying on the transient user feature.
+
+Once you have created the user, configure a JWT secret in nuxeo.conf:
+
+```
+nuxeo.jwt.secret=abracadabra
+```
+
+Then, to use the Management REST API:
+
+- share the JWT secret (`abracadabra` here) between the Nuxeo Server and the client calling the Management REST API
+- generate a JWT token with the user (`transient/technical_user` here) as claim subject
+- call the API using the `Authorization: Bearer JWT_TOKEN` header
+
+### Deploy the Management REST API on a separate port
```

Comment:
Not needed.

Reply:
then we have ```share the JWT secret (`abracadabra` here) between the Nuxeo Server and the client calling the Management REST API```
alone in a list so we can do:

> Then, to use the Management REST API, share the JWT secret (`abracadabra` here) between the Nuxeo Server and the client calling the Management REST API

Reply:
This title needs to be deleted, as we have the same title "Deploy the Management REST API on a separate HTTP port" 2 lines below.

Reply:
I have removed one of them in the process.
This is what I end up with:

> Once you have created the user, configure a JWT secret in nuxeo.conf:
> 
> ```
> nuxeo.jwt.secret=abracadabra
> ```
> 
> Then, to use the Management REST API, share the JWT secret (`abracadabra` here) between the Nuxeo Server and the client calling the Management REST API
> 
> ### Deploy the Management REST API on a separate HTTP port
> 
> For security reasons, it is recommended to deploy the Management REST API on a different port from the regular Nuxeo application one. For instance, see below to configure the HTTP port to `9090`.
> 
> #### Configure the Management REST API HTTP port
> 
> In `nuxeo.conf`, add:
> ```
> nuxeo.server.http.managementPort=9090
> ```

Reply:
Why are you removing the following:

"""
- generate a JWT token with the user (`transient/technical_user` here) as claim subject
- call the API using the `Authorization: Bearer JWT_TOKEN` header
"""
?

Reply:
Missunderstanding. Added back with the list and ":" above so all that is gone is the double title

------------------------

Then, to use the Management REST API: 
- generate a JWT token with the user (`transient/technical_user` here) as claim subject
- call the API using the `Authorization: Bearer JWT_TOKEN` header
- share the JWT secret (`abracadabra` here) between the Nuxeo Server and the client calling the Management REST API

### Deploy the Management REST API on a separate HTTP port

For security reasons, it is recommended to deploy the Management REST API on a different port from the regular Nuxeo application one. For instance, see below to configure the HTTP port to `9090`.

#### Configure the Management REST API HTTP port